### PR TITLE
CI: check if docs need to be regenerated

### PR DESCRIPTION
**Pull request by Integralist**
If a change is made to a schema `Description` field and a PR is opened, then we would like to be able to identify to the author of the PR that the documentation needs to be regenerated (e.g. by running `make generate-docs`).

The solution is to have a Github workflow which runs the relevant Makefile target and then runs `git diff` to see if there's any changes to what's in the main branch. If there are changes, it means new documentation files were generated in CI and so the author of the PR should run the relevant Makefile target locally and commit those changes.

### CODE REVIEW

**Comment by Integralist** [comment]
@phamann @doramatadora I've fixed to the latest release of goreleaser now. Ready for another review whenever you have a moment.

**Comment by Integralist** [comment]
@doramatadora @phamann ready for review.

File: GNUmakefile [hunk 1]
```diff
@@ -52,10 +52,18 @@ test-compile:
...
 	go test -c $(TEST) $(TESTARGS)
 
-generate-docs:
+dependencies:
+	@echo "Download go.mod dependencies"
+	@go mod download
+
+install-tools: dependencies
+	@echo "Installing tools from tools/tools.go"
+	@cat tools/tools.go | grep _ | awk -F '"' '{print $$2}' | xargs -tI {} go install {}
```
Comment (phamann) [inline GNUmakefile:61]: Out of interest, why have you chosen this route instead of a bare`go install`?

**Review by phamann — APPROVED** [review]
👍🏻 LGTM - my comment is just for curiosity and not blocking.

File: GNUmakefile (hunk 1 above)
Comment (Integralist) [reply GNUmakefile:61]: To clarify: `go install` needs a repo path e.g. `github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs` and so I can’t do something like `go install tools/tools.go`.

This means I needed to parse whatever is in the `tools/tools.go` file and grab all the imports. 

Admittedly there’s only one import so yeah I could just import it manually via the Makefile like so `go install github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs` but I didn’t want to couple the Makefile to a dependency that may not exist in future, nor have to update it every time a new tool is added.

**Review by doramatadora — APPROVED** [review]
LGTM 🚀

File: GNUmakefile (hunk 1 above)
Comment (phamann) [reply GNUmakefile:61]: Yup, apologies, my understanding of `go install` was incorrect :)

**Pull request merged**